File: code/python/bridge/Television.py
from Device import Device

class Television(Device):
    def __init__(self):
        self.is_on = False
        self.volume = 50
        self.channel = 1

    def power_on(self):
        self.is_on = True
        print("Television powered on")

    def power_off(self):
        self.is_on = False
        print("Television powered off")

    def adjust_volume(self, volume):
        self.volume = volume
        print(f"Television volume set to {volume}")

    def change_channel(self, channel):
        self.channel = channel
        print(f"Television channel changed to {channel}")

File: code/typescript/bridge/Television.ts
import { Device } from "./Device";

export class Television implements Device {
    private isOn: boolean;
    private volume: number;
    private channel: number;

    constructor() {
        this.isOn = false;
        this.volume = 50; // Default volume
        this.channel = 1; // Default channel
    }

    public powerOn(): void {
        this.isOn = true;
        console.log("Television powered on");
    }

    public powerOff(): void {
        this.isOn = false;
        console.log("Television powered off");
    }

    public adjustVolume(volume: number): void {
        this.volume = volume;
        console.log(`Adjusting television volume to ${volume}`);
    }

    public changeChannel(channel: number): void {
        this.channel = channel;
        console.log(`Changing television channel to ${channel}`);
    }
}
